src/components/data_comps_mct/dlnd/src/dlnd_comp_mod.F90

```diff
@@ -169,12 +170,21 @@ subroutine dlnd_comp_init(Eclock, x2l, l2x, &
     if (scmmode) then
        if (my_task == master_task) &
             write(logunit,F05) ' scm lon lat = ',scmlon,scmlat
-       call shr_strdata_init(SDLND,mpicom,compid,name='lnd', &
-            scmmode=scmmode,scmlon=scmlon,scmlat=scmlat, calendar=calendar, &
-            dmodel_domain_fracname_from_stream=domain_fracname)
+       if (domain_fracname == domain_fracname_unset) then
+          call shr_strdata_init(SDLND,mpicom,compid,name='lnd', &
+               scmmode=scmmode,scmlon=scmlon,scmlat=scmlat, calendar=calendar)
+       else
+          call shr_strdata_init(SDLND,mpicom,compid,name='lnd', &
+               scmmode=scmmode,scmlon=scmlon,scmlat=scmlat, calendar=calendar, &
+               dmodel_domain_fracname_from_stream=domain_fracname)
+       end if
     else
-       call shr_strdata_init(SDLND,mpicom,compid,name='lnd', calendar=calendar, &
-            dmodel_domain_fracname_from_stream=domain_fracname)
+       if (domain_fracname == domain_fracname_unset) then
+          call shr_strdata_init(SDLND,mpicom,compid,name='lnd', calendar=calendar)
+       else
+          call shr_strdata_init(SDLND,mpicom,compid,name='lnd', calendar=calendar, &
+               dmodel_domain_fracname_from_stream=domain_fracname)
+       end if
```

Comment:
I started down the path of trying to avoid the need for having near duplication of this code, but it was getting messy, so I decided it was simplest just to have these four different subroutine calls based on which arguments are relevant.